src/deploy/RPM_build/noobaa.spec

```diff
@@ -1,57 +1,89 @@
+# Disable debuginfo -- it does not work well with nodejs apps
+%global debug_package %{nil}
+# This speeds up the build a lot and it is not really required
+%global __os_install_post %{nil}
+
 %define revision null
 %define noobaaver null
 %define nodever null
 %define releasedate null
 %define changelogdata null
+%define BUILD_S3SELECT null
+%define BUILD_S3SELECT_PARQUET null
+%define CENTOS_VER null
+%define _build_id_links none
 
 %define noobaatar %{name}-%{version}-%{revision}.tar.gz
-%define nodetar node-%{nodever}.tar.xz
 %define buildroot %{_tmppath}/%{name}-%{version}-%{release}
 
-Name:		noobaa-core
-Version:	%{noobaaver}
-Release:	%{revision}%{?dist}
-Summary:	NooBaa RPM
+Name: noobaa-core
+Version:  %{noobaaver}
+Release:  %{revision}%{?dist}
+Summary:  NooBaa RPM
 
-License:	Apache-2.0
-URL:        https://www.noobaa.io/
-Source0:	%{noobaatar}
-Source1:    %{nodetar}
+License:  Apache-2.0
+URL:  https://www.noobaa.io/
+Source0:  %{noobaatar}
 
-BuildRequires: systemd
+BuildRequires:  systemd
+BuildRequires:  python3
+BuildRequires:  make
+BuildRequires:  gcc-c++
+BuildRequires:  boost-devel
 
-Recommends: jemalloc
-
-%global __os_install_post %{nil}
+Recommends:     jemalloc
 
 %global __requires_exclude ^/usr/bin/python$
 
 %description
 NooBaa is a data service for cloud environments, providing S3 object-store interface with flexible tiering, mirroring, and spread placement policies, over any storage resource that allows GET/PUT including S3, GCS, Azure Blob, Filesystems, etc.
 
 %prep
-mkdir noobaa-core-%{version}-%{revision}
-mkdir node-%{nodever}
-tar -xzf %{SOURCE0} -C noobaa-core-%{version}-%{revision}/
-tar -xJf %{SOURCE1} -C node-%{nodever}/
+%setup -n noobaa -q
+
+%build
+NODEJS_VERSION="%{nodever}"
+SKIP_NODE_INSTALL=1 source src/deploy/NVA_build/install_nodejs.sh $NODEJS_VERSION
+
+mkdir -p ../node/
+
+nodepath=$(download_node)
+tar -xJf ${nodepath} -C ../node/
 
-%clean
-[ ${RPM_BUILD_ROOT} != "/" ] && rm -rf ${RPM_BUILD_ROOT}
+PATH=$PATH:%{_builddir}/node/node-v$NODEJS_VERSION-linux-$(get_arch)/bin
+
+npm install --omit=dev && npm cache clean --force
+
+./src/deploy/NVA_build/clone_s3select_submodules.sh
+
+if [[ "%{CENTOS_VER}" = "8" ]]
+then
+  sed -i 's/\/lib64\/libboost_thread.so.1.75.0/\/lib64\/libboost_thread.so.1.66.0/g' ./src/native/s3select/s3select.gyp
+  echo "Using libboost 1.66 for S3 Select"
+elif [[ "%{CENTOS_VER}" = "9" ]]
+then
+  echo "Using libboost 1.75 for S3 Select"
+else
```

Comment:
I have also dropped this, the idea was that if we should not really error out unless necessary -- the build platform that we want to support in the future might not even be RHEL but SLES for instance. As such I have also removed the `elif` clause above leaving this condition RHEL/CENTOS 8 specific.

Reply:
Addressed.